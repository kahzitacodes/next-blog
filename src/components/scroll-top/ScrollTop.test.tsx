import { render, screen } from '@testing-library/react'
import { ScrollTop } from '.'

describe('<ScrollTop />', () => {
  it('should render successfully', () => {
    const { container } = render(<ScrollTop />)

    expect(
      screen.getByRole('button', { name: /voltar ao topo/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
