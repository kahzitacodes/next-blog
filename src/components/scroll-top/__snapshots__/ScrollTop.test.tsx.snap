// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollTop /> should render successfully 1`] = `
<header
  class="pointer-events-none fixed inset-0 5-50 h-full main-h-screen w-full"
>
  <button
    aria-label="voltar ao topo"
    class="pointer-events-auto absolute bottom-8 right-8 flex items-center justify-center h-12 w-12 rounded-full bg-primary/70 transition-colors duration-200 text-white hover:bg-primary [&>svg]:transition-translate [&>svg]:duration-200 [&>svg]:hover:-translate-y-0.5"
    title="voltar ao topo"
  >
    <svg
      class="lucide lucide-arrow-up"
      data-testid="ArrowUp"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m5 12 7-7 7 7"
      />
      <path
        d="M12 19V5"
      />
    </svg>
  </button>
</header>
`;
